python/cudf/cudf/core/frame.py

```diff
@@ -573,6 +573,29 @@ def equals(self, other, **kwargs):
         else:
             return self._index.equals(other._index)
 
+    def _explode(self, explode_column: Any, ignore_index: bool):
+        """Helper function for `explode` in Series and Dataframe.
+        if the designated column to explode is non-nested, a copy
+        of the frame is returned. Otherwise, if ignore_index is
```

Comment:
I don't think this function handles the non-nested copy part anymore? I think we should also leave that as an implementation detail that we reserve the right to change in the top level APIs.